grpcurl.go

```diff
@@ -161,6 +163,36 @@ func MetadataFromHeaders(headers []string) metadata.MD {
 	return md
 }
 
+var envVarRegex = regexp.MustCompile(`\${\w+}`)
+
+// ExpandHeaders expands environmental variables contained in the header string.
+// If no corresponding environmental variable is found an error is thrown.
+// TODO: Add escaping for `${`
+func ExpandHeaders(headers []string) ([]string, error) {
+	expandedHeaders := make([]string, len(headers))
+	for idx, header := range headers {
+		if header != "" {
+			results := envVarRegex.FindAllString(header, -1)
+			if results != nil {
+				expandedHeader := header
+				for _, result := range results {
+					envVarName := result[2 : len(result)-1]
+					envVarValue := os.Getenv(envVarName)
+					replacementValue := envVarValue
+					if len(envVarValue) == 0 {
```

Comment:
I think this is best tested using `os.LookupEnv` instead of `os.GetEnv`. That way, an explicitly blank environment variable is allowed -- but an _absent_ variable will result in an error:

```go
envVarValue, ok := os.LookupEnv(envVarName)
if !ok {
  return nil, fmt.Errorf("no variable %q is present in environment", envVarName)
}
```